### PR TITLE
add methods to #SodaMachine.rb to make test in provided spec pass.

### PR DESCRIPTION
**Pull request by jyonah**

### CODE REVIEW

**Comment by amgando** [comment]
nice work.  looks fine

File: soda_machine.rb [hunk 1]
```diff
@@ -7,12 +7,20 @@ def initialize(args = {})
...
   end
 
   def find_soda(soda_brand)
+    @sodas.find{|soda| soda.brand == soda_brand}
   end
 
   def sell(soda_brand)
+    this_soda = find_soda(soda_brand)
+    if this_soda != nil
+      @sodas.delete_at(@sodas.index(this_soda))
```
Comment (amgando) [inline soda_machine.rb:20]: i think this line could just be `@sodas.delete(this_soda)`

Comment (jyonah) [reply soda_machine.rb:20]: Hi @amgando!

Thanks again for the timely code review. Below is my reasoning for using #delete_at instead of #delete - please do let me know if there are any flaws in my logic. :-)

According to the [ruby docs](http://ruby-doc.org/core-2.3.1/Array.html#method-i-delete), #delete would remove all instances of this_soda, and not just one. Our [spec file](https://github.com/sf-otters-2017/phase-1-checkpoint-challenge-2-soda-machine/blob/master/spec/soda_machine_spec.rb) initializes with two instances of pepsi, so I wanted to avoid writing code that would empty the machine of pepsi altogether if a user bought a pepsi. (Ironically, the spec file doesn't actually test for purchasing Pepsi, which is the _only_ soda that was initialized twice.)

I'm pretty confident that `@sodas.delete_at(@sodas.index(this_soda))` accounts for the issue of multiple sodas - but again, we never tested for it.

Comment (amgando) [reply soda_machine.rb:20]: well, that makes perfect sense.  good catch.  why not add a test for this and submit a pull request to the challenge writers (@b0uma, @mashalifshin, @salogel42)?

that way students who take this in the future will be forced to consider the difference between these methods

Comment (jyonah) [reply soda_machine.rb:20]: @amgando (@b0uma, @mashalifshin, @salogel42)

Done and [done](https://github.com/Devbootcamp/phase-1-checkpoint-challenge-2-soda-machine/pull/12)

Comment (salogel42) [reply soda_machine.rb:20]: This is a nice addition to the test suite, but I just wanted to point out that @amgando 's suggested change would still work given the new test! Because `#delete` checks for equality, and `pespi != second_pepsi` in your test (they are different objects).  

It would be pretty weird to have the same soda object in your machine twice -- you might have multiple cans of the same *kind* of soda, but never multiple of the exact same can; that wouldn't model the real world correctly.

Comment (jyonah) [reply soda_machine.rb:20]: (@amgando) @salogel42 Oh goodness me. You're right. I'm almost a bit embarrassed now. :flushed: 

I tripped up when considering how different strings have different object ids even if they are the same string - yet they all `'a'` strings get deleted from `arr = ['a','a','b','a']` by `arr.delete('a')`. On reflection this a false equivalency because I'm comparing different STRING objects with a value of 'a' to Soda objects with their own distinct values. 

In short, I'd gotten in the habit of deleting by index likely because there was a point _at some time_ where it was somehow necessary and incorrectly learned a blanket rule. #LessonLearned

**Comment by amgando** [comment]
that's too bad, that you're feeling embarrassed. I know the feeling, when you think everything's just fine and then suddenly you realize that you missed something. well, it's a good thing that you're not expected to be perfect! and that there's no "Official Permanent Record of Mistakes You Made" being kept anywhere :)

just enjoy yourself. are you learning? did you learn something new here? great! go share it.
